File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoBlockProcessor.java
package org.wordpress.android.ui.posts.mediauploadcompletionprocessors;

import androidx.annotation.NonNull;

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.wordpress.android.util.helpers.MediaFile;

public class VideoBlockProcessor extends BlockProcessor {
    public VideoBlockProcessor(@NonNull String localId, @NonNull MediaFile mediaFile) {
        super(localId, mediaFile);
    }

    @Override
    public boolean processBlockContentDocument(@NonNull Document document) {
        // select video element with our local id
        Element targetVideo = document.select("video").first();

        // if a match is found for video, proceed with replacement
        if (targetVideo != null) {
            // replace attribute
            targetVideo.attr("src", remoteUrl);

            // return injected block
            return true;
        }

        return false;
    }

    @Override
    public boolean processBlockJsonAttributes(@NonNull JsonObject jsonAttributes) {
        JsonElement id = jsonAttributes.get("id");
        if (id != null && !id.isJsonNull() && id.getAsString().equals(localId)) {
            addIntPropertySafely(jsonAttributes, "id", remoteId);
            return true;
        }
        return false;
    }
}

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/VideoBlockProcessor.kt
package org.wordpress.android.ui.posts.mediauploadcompletionprocessors

import com.google.gson.JsonObject
import org.jsoup.nodes.Document
import org.wordpress.android.util.helpers.MediaFile

class VideoBlockProcessor(localId: String, mediaFile: MediaFile) : BlockProcessor(localId, mediaFile) {
    override fun processBlockContentDocument(document: Document): Boolean {
        // select video element with our local id
        val targetVideo = document.select("video").first()

        // if a match is found for video, proceed with replacement
        return targetVideo?.let {
            // replace attribute
            targetVideo.attr("src", remoteUrl)

            // return injected block
            true
        } ?: false
    }

    override fun processBlockJsonAttributes(jsonAttributes: JsonObject): Boolean {
        val id = jsonAttributes["id"]
        return if (id != null && !id.isJsonNull && id.asString == localId) {
            addIntPropertySafely(jsonAttributes, "id", remoteId)
            true
        } else {
            false
        }
    }
}
